libraries/AP_Math/AP_Math.h

```diff
@@ -425,9 +425,26 @@ T usec_to_hz(const T &usec) {
 }
 
 /*
-  linear interpolation based on a variable in a range
+ * @brief: linear interpolation based on a variable in a range
  */
-float linear_interpolate(float low_output, float high_output,
-                         float var_value,
-                         float var_low, float var_high);
+template <class T>
+T linear_interpolate(const T &low_output, const T &high_output,
+                     const T &var_value,
+                     const T &var_low, const T &var_high)
```

Comment:
Checking with nm confirms that with your version the function is in fact inlined